eng/targets/Imports.targets

```diff
@@ -4,6 +4,16 @@
   <Import Project="VisualStudio.targets"/>
   <Import Project="XUnit.targets" Condition="'$(IsTestProject)' == 'true'" />
 
+  <!--
+    Work around .editorconfig evaluation bugs in command line builds.
+    https://github.com/dotnet/roslyn/issues/43371
+  -->
+  <ItemGroup Condition="'$(DesignTimeBuild)' != 'true'">
+    <_AbsoluteCompile Include="@(Compile->'%(FullPath)')" />
+    <Compile Remove="@(Compile)" />
+    <Compile Include="@(_AbsoluteCompile)" />
+  </ItemGroup>
```

Comment:
@dotnet/roslyn-infrastructure please review this workaround for #43371

Reply:
Looks good to me.

Reply:
@agocke  and @chsienki need to look at this. At a glance it looks okay.

Reply:
LGTM, although this indicates to me that there's no compiler bug here, but some sort of MSBuild targets problem.

Reply:
The actual change seems fine, I'm curious to understand the underlying bug though. Can we not  just look at the full name directly inside the compiler?